src/common/row_operations/row_match.cpp

```diff
@@ -230,6 +230,29 @@ static void TemplatedMatchList(Vector &col, Vector &rows, SelectionVector &sel,
 	}
 }
 
+template <class OP, bool NO_MATCH_SEL>
+static void TemplatedMatchArray(Vector &col, Vector &rows, SelectionVector &sel, idx_t &count,
```

Comment:
I just reworked this file in #8979 but resolving the merge conflicts should be easy. You may want to add a test to `test/sql/join/inner/test_join_is_distinct.test` to double-check that everything works for `ARRAY` too.